### DISCUSSION TITLE
[Bug]: "Visual Studio Code is unable to watch for file changes in this large workspace" (error ENOSPC)

### DISCUSSION THREAD

**Discussion by YueMiyuki** [coder/code-server#4796] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS/Web Information

```markdown
- Web Browser: Opera
- Local OS: Synology DSM
- Remote OS: Windows 10
- Remote Architecture: ARM64 v8
- `code-server --version`: 4.0.1
```

### Expected

It should not return error

### Actual

It show error
```
"Visual Studio Code is unable to watch for file changes in this large workspace" (error ENOSPC)
```

### Logs

none

### Screenshot/Video

![image](https://user-images.githubusercontent.com/76854136/150637964-fa87d3e3-8225-4ae3-9e6d-b6230c5cfff2.png)

### Does this issue happen in VS Code?

- [X] I cannot reproduce this in VS Code.

### Are you accessing code-server over HTTPS?

- [X] I am using HTTPS.

### Notes

I am using the fixes at https://gist.github.com/tbjgolden/c53ca37f3bc2fab8c930183310918c8c and https://code.visualstudio.com/docs/setup/linux#_visual-studio-code-is-unable-to-watch-for-file-changes-in-this-large-workspace-error-enospc
But none of these works, it returns:
![image](https://user-images.githubusercontent.com/76854136/150637994-b08f98c6-2dd1-4c4c-ae9f-2adeb17ffd6f.png)

**Comment by jsjoeio** [comment]
Oops. Looks like our new template is missing reproduction steps. How do I reproduce this?

**Comment by YueMiyuki** [comment]
Hi
Just create an new code-server and if you have big files like `node_modules` and the project is running somewhere else,
It will return this

**Comment by mauri-sfdc** [comment]
@ItzMiracleOwO is the output of `df -i` in the terminal? are you running code-server in docker in Synology DSM?

**Comment by YueMiyuki** [comment]
yes im on DSM

![](https://cdn.discordapp.com/attachments/853150388810547211/935661217060618291/IMG_9058.png)

**Comment by jsjoeio** [comment]
Hmm...I guess this is specific to Linux. I couldn't reproduce locally on macOS.

![image](https://user-images.githubusercontent.com/3806031/151070521-b043409b-cc84-42b5-8739-c714dc1058ca.png)

I'm seeing the "permission denied" errors in your logs. Sounds like that solution isn't working. 

To me, this seems like an issue specific to your environment and not a bug in code-server. Do you agree? (Still willing to help but want to clarify and see if we're on the same page)

**Comment by YueMiyuki** [comment]
Hmm I do agree
Synology have lots of stupid issues and their docker version is ancient…
What is interesting is, before when I was in older version, this error didnt pop up..

**Comment by jsjoeio** [comment]
Hmm...well things definitely changed in code-server between versions so not ruling that out, but at least we agree. Let's move this to a Q and A then and continue there.

**Reply by jsjoeio** [reply]
Did you find any solutions online for the "permission denied" error?

**Reply by jsjoeio** [reply]
Found these: 
- https://stackoverflow.com/questions/58189187/how-to-solve-system-limit-for-number-of-file-watchers-reached-issue-when-using
- https://askubuntu.com/questions/1216710/increase-inotify-watchers-without-using-root-priviledges
- https://www.youtube.com/watch?v=tAYWte2E0_c

**Comment by YueMiyuki** [comment]
Hmmm it seems its not a Linux issue but a SYNOLOGY one cux successfully install code server on my WSL without any error…

**Comment by YueMiyuki** [comment, marked as answer]
QUICK UPDATE everyone
So I open an issue a while ago in this repo link to issue --> https://github.com/coder/code-server/issues/4058
In the last comment i said that i add the parameter `--pid=host` and it worked
That was also a peremission error so I was thinking what would happen if i add that parameter back 

The command i ran and the output:
The peremission denied is because the particular folder is privileged
![image](https://user-images.githubusercontent.com/76854136/151259032-fdc78946-99ad-4125-a829-2abfa1e7bb17.png)

Result:
![image](https://user-images.githubusercontent.com/76854136/151259058-8a705e14-5836-4088-ba70-3e5eadf80482.png)

**Reply by jsjoeio** [reply]
Oh wow! Nice work!!!! Happy you resolved it :)

**Comment by YueMiyuki** [comment]
Hmm @jsjoeio same error happened again on latest version ：(

**Reply by jsjoeio** [reply]
Ah I'm sorry! Hmm 🤔 This is on Synology DSM (which I'm not familiar with)..

Did it get reset or something? Or are you not git ignoring a folder that should be git ignored (i.e. `node_modules`)?

We didn't change anything (that I know of) in 4.0.2 that should cause this.

**Reply by YueMiyuki** [reply]
Noo
This error pop out once I set it up and login
I havent done anything yet!

**Reply by YueMiyuki** [reply]
@jsjoeio sir

**Reply by jsjoeio** [reply]
My apologies. I'm not sure what I can do to help here :(

**Reply by mauri-sfdc** [reply]
This seems like an environment issue; Not a code-server bug. 

You are mounting /home/code/project/ from an external volume. But looking at the command line you are not running the docker container by passing the host uid+gid with the `-u "$(id -u):$(id -g)"` flags (as described in https://hub.docker.com/r/codercom/code-server); that's necessary in order to not get permission denied when the user inside the container doesn't match the owner of the files in the volume you mounted

**Reply by jsjoeio** [reply]
I think @mauri-sfdc's right! Thank you for pointing that out, I'm not as experienced with Docker

**Reply by YueMiyuki** [reply]
@jsjoeio @mauri-sfdc well nope
same thing happen

**Reply by YueMiyuki** [reply]
![image](https://user-images.githubusercontent.com/76854136/153530975-52ad6741-374e-49da-ace6-b7fcb7db9485.png)
![image](https://user-images.githubusercontent.com/76854136/153530992-0b65cf61-9c85-4932-b6ea-190be2c5d23d.png)

**Reply by YueMiyuki** [reply]
Command:
`docker run --rm -it --pid=host --name test-cs2 -p 1234:8080 -v/volume1/MiracleDS/Miracle/Bots/:/home/coder/project -u "$(id -u):$(id -g)" codercom/code-server:latest`

**Reply by mauri-sfdc** [reply]
@ItzMiracleOwO that's a different error, the Inotify limit reached, which is controlled at the kernel level. Not related with codeserver. Are you confirming that running that docker command with `codercom/code-server:latest` fails but `codercom/code-server:3.12.0` works ok?

**Reply by YueMiyuki** [reply]
Noo the version that works is 4.0.1 like the here 
https://github.com/coder/code-server/discussions/4796#discussioncomment-2054652

**Reply by YueMiyuki** [reply]
@mauri-sfdc

**Comment by asthanas** [comment]
i have deployed coder on kubernetes cluster and getting the same issue.
How can we fix the same on coder running on kubernetes ?
